nemoguardrails/library/jailbreak_detection/Dockerfile-GPU

```diff
@@ -2,29 +2,31 @@
 FROM nvidia/cuda:12.3.1-base-ubuntu20.04
 
 # Install git
-RUN apt-get update && apt-get install -y git && apt-get install gcc -y && apt-get install g++ -y && apt-get clean
+RUN apt-get update && apt-get install -y git gcc g++ && apt-get clean
 
 # Install python
 RUN apt-get install python3 python3-pip python3-dev -y
 
 # Upgrade pip
-RUN pip install --upgrade pip
-
-# Copy the source code
-COPY . /app
+RUN pip install --no-cache-dir --upgrade pip
 
 # Set working directory
 WORKDIR /app
 
+# Copy the source code
+COPY requirements.txt .
+
 # Install the minimal set of requirements for jailbreak detection Server
-RUN pip install -r requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Predownload the GPT2 model.
+RUN python3 -c "from transformers import GPT2LMHeadModel, GPT2TokenizerFast; GPT2LMHeadModel.from_pretrained('gpt2-large'); GPT2TokenizerFast.from_pretrained('gpt2-large');"
```

Comment:
Here I had to change `python` to `python3`, as the image could not find the executable for the former.